payroll-processor-client/src/app/employee/employee-create/employee-create.service.ts

```diff
@@ -0,0 +1,50 @@
+import { HttpClient } from '@angular/common/http';
+import { Injectable } from '@angular/core';
+import { throwError } from 'rxjs';
+import { catchError, tap } from 'rxjs/operators';
+
+import { EmployeeCreate, EmployeeCreateResponse } from '../employee-list/state/employee-list.model';
+import { EmployeeListStore } from '../employee-list/state/employee-list.store';
+
+import { EnvService } from 'src/app/shared/env.service';
+
+@Injectable({ providedIn: 'root' })
+export class EmployeeCreateService {
+  private readonly apiUrl: string;
+
+  constructor(
+    private http: HttpClient,
+    private employeeListStore: EmployeeListStore,
+    // private router: Router,
+    envService: EnvService,
+  ) {
+    this.apiUrl = envService.apiRootUrl;
+  }
+
+  createEmployee(employee: EmployeeCreate) {
+    // console.log(employee);
+    return this.http
+      .post<EmployeeCreateResponse>(`${this.apiUrl}/employees`, employee)
+      .pipe(
+        tap((detail) => {
+          this.employeeListStore.upsert(detail.rowKey, {
+            id: detail.rowKey,
```

Comment:
In this case `detail` will be `Employee`, not `EmployeeEntity`, so it will have an `id` and not a `rowKey`

Reply:
Yeah this was something i changed to make sure my store update and subscription was working, then as we made design changes i just needed to push up.